Patch-Source: https://github.com/facebookincubator/below/pull/8162
--
From f5d58403d719d3820c2bef94f6d01b5727f246c3 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 14 Oct 2022 21:02:39 +0200
Subject: [PATCH] Allow to build without vendored libraries

Linux distributions don't like bundling dependencies, for good reasons.
This commit adds a new feature flag "no-vendor" which disables using
vendored libbpf and libzstd (i.e. the binary is linked with the
system-provided libraries) when enabled.
---
 below/Cargo.toml       | 1 +
 below/store/Cargo.toml | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/below/Cargo.toml b/below/Cargo.toml
index 1b245341..2a4584a3 100644
--- a/below/Cargo.toml
+++ b/below/Cargo.toml
@@ -47,3 +47,4 @@ libbpf-cargo = "0.12.0"
 
 [features]
 enable_backtrace = []
+no-vendor = ["libbpf-cargo/novendor", "libbpf-rs/novendor", "store/no-vendor"]
diff --git a/below/store/Cargo.toml b/below/store/Cargo.toml
index 2155302c..110be63c 100644
--- a/below/store/Cargo.toml
+++ b/below/store/Cargo.toml
@@ -32,3 +32,6 @@ paste = "1.0"
 slog-term = "2.8"
 tempdir = "0.3"
 zstd = "0.11.1+zstd.1.5.2"
+
+[features]
+no-vendor = ["zstd/pkg-config", "zstd-safe/pkg-config"]
